src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -1175,17 +1161,195 @@ def height(self):
+            sage: psi.is_isomorphic(phi)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: cannot solve the equation u^24 == T
+
+        However, it never fails over the algebraic closure::
+
+            sage: psi.is_isomorphic(phi, absolutely=True)
+            True
+
+        Note finally that when the constant coefficients of `\phi_T` and
+        `\psi_T` differ, `\phi` and `\psi` do not belong to the same category
+        and checking whether they are isomorphic does not make sense; in this
+        case, an error is raised::
+
+            sage: phi = DrinfeldModule(A, [z, 0, 1])
+            sage: psi = DrinfeldModule(A, [z^2, 0, 1])
+            sage: phi.is_isomorphic(psi)
+            Traceback (most recent call last):
+            ...
+            ValueError: Drinfeld modules are not in the same category
+
+        TESTS:
+
+        A Drinfeld module is always isomorphic to itself::
+
+            sage: phi = DrinfeldModule(A, [z] + [K.random_element() for _ in range(3)] + [1])
+            sage: phi.is_isomorphic(phi)
+            True
+
+        Two Drinfeld modules of different ranks are never isomorphic::
+
+            sage: psi = DrinfeldModule(A, [z] + [K.random_element() for _ in range(5)] + [1])
+            sage: phi.is_isomorphic(psi)
+            False
+
+        Two Drinfeld modules which are not patterned-alike are also not
+        isomorphic::
+
+            sage: phi = DrinfeldModule(A, [T, 1, 0, 1])
+            sage: psi = DrinfeldModule(A, [T, 1, 1, 1])
+            sage: phi.is_isomorphic(psi)
+            False
+        """
+        if self.category() is not other.category():
+            raise ValueError("Drinfeld modules are not in the same category")
+        if self is other:
+            return True
+        r = self.rank()
+        if other.rank() != r:
+            return False
+        q = self._Fq.cardinality()
+        A = self.gen()
+        B = other.gen()
+        e = Integer(0)
+        ue = self._base(1)
+        for i in range(1, r+1):
+            ai = A[i]
+            bi = B[i]
+            if ai == 0 and bi == 0:
+                continue
+            if ai == 0 or bi == 0:
+                return False
+            if e != q - 1:
+                # u^e = ue
+                # u^(q^i - 1) = ai/bi
+                e, s, t = e.xgcd(q**i - 1)
+                ue = ue**s * (ai/bi)**t
+        for i in range(1, r+1):
+            if A[i]:
+                f = (q**i - 1) // e
+                if A[i] != B[i] * ue**f:
+                    return False
+        if absolutely:
+            return True
+        else:
+            ue = ue.backend(force=True)
+            try:
+                _ = ue.nth_root(e)
+            except ValueError:
+                return False
+            except (AttributeError, NotImplementedError):
+                raise NotImplementedError("cannot solve the equation u^%s == %s" % (e, ue))
```

Comment:
More consistent with the codebase of Drinfeld modules:

```suggestion
                raise NotImplementedError(f"cannot solve the equation u^{e} == {ue}")
```